init.lua

```diff
@@ -14,41 +14,20 @@ local SUPPORTED_KEYS = {
 	{ on = "u"}, { on = "v"}, { on = "w"}, { on = "x"}, { on = "y"}, { on = "z"},
 }
 
-local function save_bookmark(idx)
-	if idx == -1 then
-		return
-	end
-
-	local folder = Folder:by_kind(Folder.CURRENT)
-
-	local key = SUPPORTED_KEYS[idx].on
-
+local save_bookmark = ya.sync(function(idx)
 	state.bookmarks = state.bookmarks or {}
-	state.bookmarks[key] = { cursor = folder.cursor, path = tostring(folder.cwd) }
-end
+	state.bookmarks[#state.bookmarks + 1] = {
+		on = SUPPORTED_KEYS[idx].on,
+		desc = tostring(folder.cwd),
```

Comment:
If I comment out this line the changes work for me, so there's no problem with the `state`, I was probably doing something wrong yesterday.
The issue on my end is just that `folder` is `nil`

Reply:
Hi, I made a new PR https://github.com/dedukun/bookmarks.yazi/pull/3, it should fix the `folder`

Reply:
Is that `folder` a global variable? Or should it just be `Folder:by_kind(Folder.CURRENT).cwd`?

Reply:
No, that's a typo. It should be: `local folder = Folder:by_kind(Folder.CURRENT)`